File: src/main/java/spring/webmvc/presentation/dto/request/MemberSaveRequest.java
package spring.webmvc.presentation.dto.request;

import java.time.LocalDate;
import java.util.List;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;

public record MemberSaveRequest(
	@NotBlank
	@Email
	String account,
	@NotBlank
	String password,
	@NotBlank
	String name,
	@Pattern(regexp = "^010-\\d{3,4}-\\d{4}$")
	String phone,
	@NotNull
	LocalDate birthDate,
	List<Long> roleIds,
	List<Long> permissionIds
) {
	public MemberSaveRequest {
		if (roleIds == null) {
			roleIds = List.of();
		}

		if (permissionIds == null) {
			permissionIds = List.of();
		}
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/dto/request/MemberSaveRequest.kt
package spring.web.kotlin.presentation.dto.request

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.Pattern
import java.time.LocalDate


data class MemberSaveRequest(
    @field:Email
    val account: String,
    val password: String,
    val name: String,
    @field:Pattern(regexp = "^010-\\d{3,4}-\\d{4}$")
    val phone: String,
    val birthdate: LocalDate,
    val roleIds: List<Long> = emptyList(),
    val permissionIds: List<Long> = emptyList(),
)
